zproject/test_extra_settings.py

```diff
@@ -201,9 +202,24 @@ def set_loglevel(logger_name: str, level: str) -> None:
 # By default two factor authentication is disabled in tests.
 # Explicitly set this to True within tests that must have this on.
 TWO_FACTOR_AUTHENTICATION_ENABLED = False
-PUSH_NOTIFICATION_BOUNCER_URL: str | None = None
 DEVELOPMENT_DISABLE_PUSH_BOUNCER_DOMAIN_CHECK = False
 
+# Disable all services by default. Tests can activate them by overriding
+# settings explicitly when they want to enable something.
+# We set this to None instead of an empty set to get the same state
+# as a server that doesn't uncomment its ZULIP_SERVICES line in settings.py.
+ZULIP_SERVICE_PUSH_NOTIFICATIONS = False
+ZULIP_SERVICE_SUBMIT_USAGE_STATISTICS = False
+ZULIP_SERVICE_SECURITY_ALERTS = False
```

Comment:
Some small tweaks to the test code, and I'll also push documentation changes.

``` diff
diff --git a/zerver/lib/test_helpers.py b/zerver/lib/test_helpers.py
index 001fe70b4d..a62ab66cfd 100644
--- a/zerver/lib/test_helpers.py
+++ b/zerver/lib/test_helpers.py
@@ -99,7 +99,7 @@ class activate_push_notification_service(override_settings):  # noqa: N801
         assert zulip_services_url is not None
 
         # Ordinarily the ANALYTICS_DATA_UPLOAD_LEVEL setting is computed based on these
-        # ZULIP_SERVIICE_* configured settings; but because these settings here won't get
+        # ZULIP_SERVICE_* configured settings; but because these settings here won't get
         # processed through computed_settings, we need to set ANALYTICS_DATA_UPLOAD_LEVEL
         # manually.
         # The logic here is:
diff --git a/zproject/test_extra_settings.py b/zproject/test_extra_settings.py
index 55ad895fdc..af455a3865 100644
--- a/zproject/test_extra_settings.py
+++ b/zproject/test_extra_settings.py
@@ -204,10 +204,9 @@ BIG_BLUE_BUTTON_URL = "https://bbb.example.com/bigbluebutton/"
 TWO_FACTOR_AUTHENTICATION_ENABLED = False
 DEVELOPMENT_DISABLE_PUSH_BOUNCER_DOMAIN_CHECK = False
 
-# Disable all services by default. Tests can activate them by overriding
-# settings explicitly when they want to enable something.
-# We set this to None instead of an empty set to get the same state
-# as a server that doesn't uncomment its ZULIP_SERVICES line in settings.py.
+# Disable all Zulip services by default. Tests can activate them by
+# overriding settings explicitly when they want to enable something,
+# often using activate_push_notification_service.
 ZULIP_SERVICE_PUSH_NOTIFICATIONS = False
 ZULIP_SERVICE_SUBMIT_USAGE_STATISTICS = False
 ZULIP_SERVICE_SECURITY_ALERTS = False
```